lib/Solver/Z3Solver.cpp

```diff
@@ -261,6 +261,10 @@ bool Z3SolverImpl::internalRunSolver(
   Z3ASTHandle z3QueryExpr =
       Z3ASTHandle(builder->construct(query.expr), builder->ctx);
 
+  for(auto& arrayIndexValueExpr : builder->constant_array_assertions) {
+      Z3_solver_assert(builder->ctx, theSolver, arrayIndexValueExpr);
```

Comment:
This isn't good. You'll be adding these assertions, even when are completely irrelevant to the current query.